src/test/kotlin/graphics/scenery/tests/examples/advanced/VideoDecodingExample.kt

```diff
@@ -0,0 +1,103 @@
+package graphics.scenery.tests.examples.advanced
+
+import graphics.scenery.BufferUtils
+import graphics.scenery.DetachedHeadCamera
+import graphics.scenery.FullscreenObject
+import graphics.scenery.SceneryBase
+import graphics.scenery.backends.Renderer
+import graphics.scenery.textures.Texture
+import graphics.scenery.utils.VideoDecoder
+import org.joml.Quaternionf
+import org.joml.Vector3f
+import org.joml.Vector3i
+import java.nio.ByteBuffer
+import kotlin.concurrent.thread
+import kotlin.test.assertTrue
+
+/**
+ * Example to show programmatic video decoding. It also demonstrates how the [FullscreenObject] class and its associated shaders may
+ * be used to display an image in full screen.
+ *
+ * @author Aryaman Gupta <argupta@mpi-cbg.de>
+ */
+class VideoDecodingExample : SceneryBase("VideoDecodingExample", 600, 600, wantREPL = false) {
+
+    var buffer: ByteBuffer = ByteBuffer.allocateDirect(0)
+    var cnt: Int = 0
```

Comment:
Changed the variable name. It does need to be a class property because its used in the `init` function as well as the `main` function for the assertion